backend/api/Areas/Admin/Controllers/AgencyController.cs

```diff
@@ -137,11 +157,18 @@ public IActionResult UpdateAgency([FromBody] Model.AgencyModel model)
         [ProducesResponseType(typeof(Model.AgencyModel), 200)]
         [ProducesResponseType(typeof(Api.Models.ErrorResponseModel), 400)]
         [SwaggerOperation(Tags = new[] { "admin-agency" })]
-        public IActionResult DeleteAgency([FromBody] Model.AgencyModel model)
+        public async Task<IActionResult> DeleteAgencyAsync([FromBody] Model.AgencyModel model)
         {
             var entity = _mapper.Map<Entity.Agency>(model);
             _pimsAdminService.Agency.Remove(entity);
 
+            // TODO: This isn't ideal as the db update may be successful but this request may not.
```

Comment:
Need to tell Keycloak to update the users that were affected by the agency change.